From 2d65fa82859c48bb97e1a6113e8cd79db3ac36e0 Mon Sep 17 00:00:00 2001
From: Will Moss <willormos@gmail.com>
Date: Thu, 16 Mar 2023 17:45:46 +0530
Subject: [PATCH 068/500] OpenWrt:
 684-of_net-do-mac-address-increment-only-once.patch

of_net: do mac-address-increment only once

Remove mac-address-increment and mac-address-increment-byte
DT property after incrementing process to make sure MAC address
would not get incremented more if this function is stared again.
It could happen if device initialization is deferred after
unsuccessful attempt.

Signed-off-by: Will Moss <willormos@gmail.com>

Change-Id: Ibc58e9d97b1e8c2337f911b508ec2ce86347e1c7
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 1bd85e0413a950385efc8497115bc2c0bebdd67c)
---
 net/core/of_net.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/net/core/of_net.c b/net/core/of_net.c
index e40da6030ecf..1cc80a2dc063 100644
--- a/net/core/of_net.c
+++ b/net/core/of_net.c
@@ -196,6 +196,12 @@ int of_get_mac_address(struct device_node *np, u8 *addr)
 		addr[3] = (mac_val >> 16) & 0xff;
 		addr[4] = (mac_val >> 8) & 0xff;
 		addr[5] = (mac_val >> 0) & 0xff;
+
+		/* Remove mac-address-increment and mac-address-increment-byte
+		 * DT property to make sure MAC address would not get incremented
+		 * more if this function is stared again. */
+		of_remove_property(np, of_find_property(np, "mac-address-increment", NULL));
+		of_remove_property(np, of_find_property(np, "mac-address-increment-byte", NULL));
 	}
 
 	of_add_mac_address(np, addr);
-- 
2.34.1

